docs/code-style.md

```diff
@@ -0,0 +1,667 @@
+```
+
+For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+    { age  :: !Int     -- ^ Age
+    , name :: !String  -- ^ First name
+    }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+    { -- | This is a very very very long comment that is split over
+      -- multiple lines.
+      field1 :: !Text
+
+      -- | This is a second very very very long comment that is split
+      -- over multiple lines.
+    , field2 :: !Int
+    }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser = Parser
+    !Int         -- Current position
+    !ByteString  -- Remaining input
+
+foo :: Int -> Int
+foo n = salt * n + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
+
+Write each `LANGUAGE` pragma on its own line, sort them alphabetically and
+align by max width among them.
+
+```haskell
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE Rank2Types       #-}
+{-# LANGUAGE TemplateHaskell  #-}
+```
+
+### Module name
+
+Use singular when naming modules (e.g. use `Data.Map` and
+`Data.ByteString.Internal` instead of `Data.Maps` and
+`Data.ByteString.Internals`). Sometimes it's acceptable to use plural
+(e. g. `Types`, `Instances`).
+
+### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+       ( -- * The @Set@ type
+         Set
+       , empty
+       , singleton
+
+         -- * Querying
+       , member
+       ) where
+```
+
+Some clarifications:
+
+1. Use 7 spaces indentation for export list (so that bracket is below
+   the first letter in module name).
+2. You can split export list into sections or just write all as single section.
+3. It is strongly adviced to sort each section alpabetically. However,
+   classes, data types and type aliases should be written before
+   functions.
+
+### Imports
+
+Imports should be grouped in the following order:
+
+0. Implicit import of custom prelude (for example [`universum`](https://github.com/serokell/universum)) if used.
+1. Everything from hackage packages or from your packages outside current project.
+2. Everything from current project.
+3. Everything from current target (like `Bench.*` or `Test.*`).
+4. Qualified imports
```

Comment:
(4) makes sense for me, although I wonder what will we do when someone qualified-imports modules from the current project.
On (3) I would like to disagree. For instance look at [this](https://github.com/DisciplinaOU/disciplina/blob/master/educator/tests/Test/Dscp/Educator/Web/Educator/Queries.hs#L11-L18), here imports are already perfectly distinguished without being separated by a blank line.

Reply:
I'd like to leave (3), because I still think that your example would look better with a blank line. Also, without a blank line test modules will be under lib modules, but bench modules will be _above_ them (because `B` < `D` < `T`, hehe). That would be irritating, so I'd leave the blank line.